tests/job-spawner-and-cleaner.py

```diff
@@ -34,66 +39,70 @@
 while doom_counter < 20:
     if counter < 256:
         # Spawn 256 initial jobs in each queue
-        for queue in range(len(queue_list)-1):
+        for queue in range(len(queue_list)):
             queue_list[queue].add_item(db, Item(bytes([counter])))
     elif counter % 2 == 0:
         # Every other tick just log how much work is left
         for queue in queue_list:
             print((queue.queue_len(db), queue.processing(db)))
+        
+        print(shared_jobs.queue_len(db),shared_jobs.processing(db))
+
         sleep(0.5)
     elif counter == 501:
         # After a little bit, add more jobs.
         print("More jobs!!")
         for n in range(0, 256):
             n = n % 256
-            for queue in range(len(queue_list)-1):
+            for queue in range(len(queue_list)):
                 queue_list[queue].add_item(db, Item(bytes([n])))
     elif doom_counter > 10 and not revived:
         # After everything settles down, add more jobs
         print("Even more jobs!!")
         revived = True
         for n in range(0, 256):
-            for queue in range(len(queue_list)-1):
+            for queue in range(len(queue_list)):
                 queue_list[queue].add_item(db, Item(bytes([n])))
     else:
         # Otherwise, clean!
         print("Cleaning")
         for queue in queue_list:
             queue.light_clean(db)
 
+        shared_jobs.light_clean(db)
     # The `doom_counter` counts the number of consecutive times all the lengths are 0.
     doom_counter = doom_counter + 1 if all(map(
         lambda queue: queue.queue_len(db) == 0 and queue.processing(db) == 0, queue_list,
-    )) else 0
+    )) and shared_jobs.processing(db) == 0 and shared_jobs.queue_len(db) == 0  else 0
```

Comment:
We could instead iterate over `queue_list + [shared_queue]` to remove the duplicated code.